### ISSUE TITLE
Major release to drop support for 14.x

### ISSUE THREAD

**Issue by mhdawson** [nodejs/node-addon-api#1315]
With Node.js 14.x going EOL today it's time to consider a major in which we'd drop support.

**Comment by mhdawson** [comment]
Discussed in team meeting today and we should also document the process for dropping support which includes

1) Updating minimum version supported in README.md
2) Removing from GitHub actions
3) Removing from Jenkins CI - https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-node-addon-api-LTS%20versions/
4) Publishing a major release

**Comment by KevinEady** [comment]
Hi team (@NickNaso, @mhdawson, @legendecas , @vmoroz , @JckXia , @gabrielschulhof )

I am working on the release and following [Creating a Release](https://github.com/nodejs/node-addon-api/blob/main/doc/creating_a_release.md). One thing I have a concern about is the final updates to `package.json`, `CHANGELOG.md`, `README.md` prior to NPM publishing.

It looks like for previous releases, these changes were committed directly into `main`: 

- [v6.1.0](https://github.com/nodejs/node-addon-api/commit/0e34f228399b7ef7b2223e8320d8260e6a707e03)
- [v6.0.0](https://github.com/nodejs/node-addon-api/commit/c51d6b321fa4fbbc134630ce0c72bdf9a2664b2d)
- [v5.1.0](https://github.com/nodejs/node-addon-api/commit/34221c1fe69bbbe4762a6a3bede80456327dd3c2)

I believe these changes should be introduced as a PR. The current workflow with editing `main` directly only works because we do not have branch protection (which we _should_ have, according to [OpenSSF scorecard](https://deps.dev/project/github/nodejs%2Fnode-addon-api)).

I would like to merge #1324 , then create another PR to perform these final updates. Once that is merged into main, then continue the release/publishing. 

How does this sound?

**Comment by vmoroz** [comment]
@KevinEady, good point! It is better to follow the best practices regarding security.

**Comment by KevinEady** [comment]
CI for release on https://github.com/nodejs/node-addon-api/commit/39a25bf27788ff7a7ea5c64978c4dcd1e7b9d80d:
| Version | Job | Status |
|---:|---|:---:|
| main | https://ci.nodejs.org/job/node-test-node-addon-api-new/7586/ |  |
| v20.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/7585/ |  |
| v18.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/7584/ |  |
| v16.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/7583/ |  |

**Comment by KevinEady** [comment]
@NickNaso @mhdawson it appears that I do not have access to publish the package on NPM. I have access to the other `node-api-header` repo (the "Collaborators" on the [node-api-header npm page](https://www.npmjs.com/package/node-api-headers) shows my name) but not on [node-addon-api npm page](https://www.npmjs.com/package/node-addon-api)

**Comment by KevinEady** [comment]
> @KevinEady what is your npm id? Just want to be sure before I add you :).

It's `kevineady`

**Comment by mhdawson** [comment]
k invited

**Comment by mhdawson** [comment]
@KevinEady  Is this still to be done ?

> I would like to merge https://github.com/nodejs/node-addon-api/pull/1324 , then create another PR to perform these final updates. Once that is merged into main, then continue the release/publishing.

As we don't seem to have a 7.0.0 release in the repo?

**Comment by KevinEady** [comment]
> Create a release in Github (look at existing releases for an example).

🤦 

This has now been completed: https://github.com/nodejs/node-addon-api/releases/tag/v7.0.0
